packages/broker/test/integration/plugins/mqtt/MqttPlugin.test.ts

```diff
@@ -0,0 +1,117 @@
+import { StreamrClient } from 'streamr-client'
+import { AsyncMqttClient } from 'async-mqtt'
+import { Wallet } from 'ethers'
+import { wait, waitForCondition } from 'streamr-test-utils'
+import { startTracker, Tracker } from 'streamr-network'
+import mqtt from 'async-mqtt'
+import { Broker } from '../../../../src/broker'
+import { createMockUser, createClient, startBroker } from '../../../utils'
+
+const MQTT_PORT = 1883
+const WS_PORT = 12391
+const TRACKER_PORT = 12392
+const NETWORK_PORT = 12393
+const MOCK_API_KEY = 'mock-api-key'
+
+const createMqttClient = () => {
+    return mqtt.connectAsync('mqtt://localhost:' + MQTT_PORT, {
+        username: '',
+        password: MOCK_API_KEY,
+    })
+}
+
+describe('MQTT plugin', () => {
+
+    let streamrClient: StreamrClient
+    let mqttClient: AsyncMqttClient
+    let user: Wallet
+    let topic: string
+    let tracker: Tracker
+    let broker: Broker
+
+    const getStreamId = () => user.address + '/' + topic
+
+    beforeAll(async () => {
+        user = createMockUser()
+        tracker = await startTracker({
+            id: 'tracker',
+            host: '127.0.0.1',
+            port: TRACKER_PORT,
+        })
+        broker = await startBroker({
+            name: 'broker',
+            privateKey: user.privateKey,
+            networkPort: NETWORK_PORT,
+            trackerPort: TRACKER_PORT,
+            wsPort: WS_PORT,
+            extraPlugins: {
+                mqtt: {
+                    port: MQTT_PORT,
+                    streamIdDomain: user.address
+                }
+            },
+            apiAuthentication: {
+                keys: [MOCK_API_KEY]
+            }
+        })
+    })
+
+    afterAll(async () => {
+        await broker.close()
+        await tracker.stop()
```

Comment:
With `allSettled` you can make sure that the if e.g. `await broker.close()` would throw then `tracker.stop()` also gets called.
```suggestion
        await Promise.allSettled([
            broker.close(),
            tracker.stop()
        ])
```

Reply:
Fixed